### PR TITLE
fix(@cubejs-backend/mssql-driver): add column type mappings for MSSQL external pre-aggregations

### PR DESCRIPTION
**Pull request by davefrancis**
Hi, this PR provides type mappings from generic types to MSSQL data types. Fixes bugs when creating external pre-aggregation tables in a MSSQL database.

**Check List**
- [ ] Tests has been run in packages where changes made if available
- [x] Linter has been run for changed code
- [ ] Tests for the changes have been added if not covered yet
- [ ] Docs have been added / updated if required

**Issue Reference this PR resolves**

#2831

### CODE REVIEW

**Comment by davefrancis** [comment]
Hi @rpaik, here is the PR regarding the MSSQL bug. I was unsure whether the `timestamp` generic type is time zone aware. If it is the PR should changed before it is merged. I also found the string and text type mappings to be incorrect as well and they are also fixed in this PR. Thanks!

**Comment by rpaik** [comment]
Thanks @davefrancis 😄 

@ovr can you help with the review?

File: packages/cubejs-mssql-driver/driver/MSSqlDriver.js [hunk 1]
```diff
@@ -1,6 +1,13 @@
 const sql = require('mssql');
 const { BaseDriver } = require('@cubejs-backend/query-orchestrator');
 
+const GenericTypeToMSSql = {
+  string: 'nvarchar(max)',
+  text: 'nvarchar(max)',
+  // should this be timezone aware? use datetimeoffset
+  timestamp: 'datetime',
```
Comment (ovr) [inline packages/cubejs-mssql-driver/driver/MSSqlDriver.js:8]: I am thinking about `datetime2`, but it's supported in MSSQL >= 2008

Comment (davefrancis) [reply packages/cubejs-mssql-driver/driver/MSSqlDriver.js:8]: @ovr I think you're right that `datetime2` would be a better option. It seems to be the suggested type from Microsoft going forward. This is from their documentation: 

> Use the time, date, datetime2 and datetimeoffset data types for new work. These types align with the SQL Standard. They are more portable. time, datetime2 and datetimeoffset provide more seconds precision. datetimeoffset provides time zone support for globally deployed applications.

I will update the source branch to use `datetime2`. Thanks for the review.

**Pull request merged**